public/app/plugins/panel/geomap/components/Tooltip.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { PureComponent } from 'react';
+import { stylesFactory } from '@grafana/ui';
+import { Map } from 'ol';
+import { GrafanaTheme } from '@grafana/data';
+import { css } from '@emotion/css';
+import { config } from 'app/core/config';
+import tinycolor from 'tinycolor2';
+import { Layer } from 'ol/layer';
+
+interface Props {
+  map: Map;
+  layerID: string;
+}
+
+interface State {
+  visible: boolean;
+  coordinates?: number[];
+}
+
+export class Tooltip extends PureComponent<Props, State> {
+  style = getStyles(config.theme);
+
+  constructor(props: Props) {
+    super(props);
+    this.state = { visible: false };
+  }
+
+  updateViewState = (onFeature: boolean, coordinates: number[]) => {
+    this.setState({
+      visible: onFeature,
+      coordinates: coordinates,
+    });
+  };
+
+  componentDidMount() {
+    const { map, layerID } = this.props;
+
+    map.on('pointermove', (evt: any) => {
+      const pixel = map.getEventPixel(evt.originalEvent);
```

Comment:
FYI this checks across all layers (we only have one so it does not really matter)